pkg/nimble/autoconn/nimble_autoconn.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (C) 2018-2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_nimble_autoconn
+ * @{
+ *
+ * @file
+ * @brief       Autoconn connection manager implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "assert.h"
+#include "random.h"
+#include "net/bluetil/ad.h"
+#include "net/bluetil/addr.h"
+#include "nimble_netif.h"
+#include "nimble_netif_conn.h"
+#include "nimble_scanner.h"
+#include "nimble_autoconn.h"
+#include "nimble_autoconn_params.h"
+
+#include "host/ble_hs.h"
+#include "nimble/nimble_port.h"
+
+#define ENABLE_DEBUG    (1)
+#include "debug.h"
+
+#if defined(MODULE_NIMBLE_AUTOCONN_IPSP)
+#define SVC_FILTER      BLE_GATT_SVC_IPSS
+#elif defined(MODULE_NIMBLE_AUTOCONN_NDNSP)
+#define SVC_FILTER      BLE_GATT_SVC_NDNSS
+#else
+#error "NimBLE autoconn: please select a fitting submodule"
+#endif
+
+#define CONN_TIMEOUT_MUL            (5U)
+
+enum {
+    STATE_SCAN,
```

Comment:
there is always documentation that could be added. But honestly, a self speaking field in a source file is not high on my priority list...

Reply:
+1

```
STATE_SCAN /**< node scanning */
STATE_ADV /**< node advertising */
STATE_IDLE /**< node is idle */
STATE_CONN /**< nide is connected */
```

That's written without context, by looking at the state names. IMO, if any state means something different (e.g., not clear from the enum name), it justifies more documentation. Otherwise it is useless.

Reply:
Ok, I was just trying to be consistent with how most of our code looks.